packages/react/src/components-shared/DefaultPositionedUiElement.tsx

```diff
@@ -0,0 +1,57 @@
+import {
+  BlockNoteEditor,
+  BlockSchema,
+  DefaultBlockSchema,
+  DefaultInlineContentSchema,
+  DefaultStyleSchema,
+  InlineContentSchema,
+  StyleSchema,
+  UiElementPosition,
+} from "@blocknote/core";
+import { flip, offset, UseFloatingOptions } from "@floating-ui/react";
+import { FC } from "react";
+import { useUiElement } from "../hooks/useUiElement";
+import { useUiElementPosition } from "../hooks/useUiElementPosition";
+
+export const DefaultPositionedUiElement = <
+  State extends UiElementPosition,
+  BSchema extends BlockSchema = DefaultBlockSchema,
+  I extends InlineContentSchema = DefaultInlineContentSchema,
+  S extends StyleSchema = DefaultStyleSchema
+>(props: {
+  editor: BlockNoteEditor<BSchema, I, S>;
+  onUpdate: (callback: (state: State) => void) => void;
+  zIndex: number;
+  options?: Partial<UseFloatingOptions>;
```

Comment:
options is not used

Reply:
This component is the generic version of the menu/toolbar specific `DefaultPositioned...` components. It's not finished bc we still need to discuss whether adding another level of abstraction here is a good idea.